src/Collection/ArraySequence.php

```diff
@@ -16,6 +16,15 @@
 
     private $array;
 
+    public static function coalesce(self $value = null) : self
```

Comment:
Maybe we should have `sequence_for()` function (similar to Guzzle's `promise_for()`) that returns the appropriate type (`PromiseSequence` for a `PromiseInterface`, `ArraySequence` for an `array` or any single value).